rotkehlchen/exchanges/woo.py

```diff
@@ -0,0 +1,541 @@
+                    case=case,
+                )
+            try:
+                response_dict: dict[str, Any] = response.json()
+                entries: list[dict[str, Any]] = response_dict['data' if case == 'trades' else 'rows']  # noqa: E501
+            except (JSONDecodeError, KeyError):
+                msg = f'Woo {case_pretty} returned an invalid JSON response: {response.text}.'
+                log.error(msg, options=call_options)
+                self.msg_aggregator.add_error(
+                    f'Got remote error while querying Woo {case_pretty}: {msg}',
+                )
+                return []
+            for entry in entries:
+                try:
+                    result = deserialization_method(entry)
+                except (DeserializationError, UnknownAsset, KeyError) as e:
+                    log.error(
+                        f'Error processing a Woo {case_pretty}.',
+                        entry=entry,
+                        error=str(e),
+                    )
+                    self.msg_aggregator.add_error(
+                        f'Failed to deserialize a Woo {case_pretty}. '
+                        f'Check logs for details. Ignoring it.',
+                    )
+                    continue
+                results.append(result)
+            if len(entries) < API_MAX_LIMIT:
+                break
+            try:
+                if case == 'trades':
+                    call_options['fromId'] = entries[-1]['id']
+                elif (meta := response_dict.get('meta')) is not None and meta['current_page'] * meta['records_per_page'] < meta['total']:  # noqa: E501
+                    call_options['page'] += 1
+                continue
+            except KeyError as e:
+                log.error(
+                    f'Error loading all Woo {case_pretty}. The entries do not have the key {e}.',
+                    entries=entries,
+                )
+                self.msg_aggregator.add_error(
+                    f'Failed to load all Woo {case_pretty}. Check logs for details.',
+                )
+            except TypeError as e:
+                log.error(
+                    f'Error loading all Woo {case_pretty}. The response keys have improper types {e}.',  # noqa: E501
+                    response=response_dict,
+                )
+                self.msg_aggregator.add_error(
+                    f'Failed to load all Woo {case_pretty}. Check logs for details.',
+                )
+            break
+        return results
+
+    def _deserialize_asset_movement(self, movement: dict[str, Any]) -> AssetMovement:
+        """
+        Deserialize a Woo asset movement returned from the API to AssetMovement
+
+        May raise:
+        - DeserializationError
+        - UnknownAsset
+        - KeyError
+        """
+        asset = asset_from_woo(movement['token'])
+        fee_currency = asset_from_woo(movement['fee_token'] if movement['fee_token'] != '' else 'USDT')  # noqa: E501
```

Comment:
where does this USDT come from? is it in the docs? if missing then it is 0?

Reply:
that's the standard in rotki I noticed for fees if there're no fees

Reply:
That is not true. If there is no fees, fee should be `None` but since the `AssetMovement` is an old structure and will go away soon this has not happened. 

If **anything** is the standard is for `fee_token` to be the same as the asset being deposited/withdrawn. Absolutely not USDT. I mean that's completely arbitrary ...